it/controllers/benefits/TravelOrSubsistenceBenefitsAmountControllerISpec.scala

```diff
@@ -0,0 +1,412 @@
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          "has an BAD_REQUEST status" in {
+            result.status shouldBe BAD_REQUEST
+          }
+
+          titleCheck(get.expectedErrorTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(user.commonExpectedResults.expectedCaption)
+          textOnPageCheck(user.commonExpectedResults.previousExpectedContent + " " + get.expectedContent, contentSelector)
+          textOnPageCheck(user.commonExpectedResults.amountHint, hintTextSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          inputFieldValueCheck("abc", inputSelector)
+
+          buttonCheck(user.commonExpectedResults.continue, continueButtonSelector)
+          formPostLinkCheck(continueLink, continueButtonFormSelector)
+          welshToggleCheck(user.isWelsh)
+
+          errorSummaryCheck(get.wrongFormatErrorText, expectedErrorHref)
+          errorAboveElementCheck(get.wrongFormatErrorText)
+        }
+
+        "should render the travel or subsistence benefits amount page with max error when input > 100,000,000,000" which {
+          implicit lazy val result: WSResponse = {
+            authoriseAgentOrIndividual(user.isAgent)
+            dropEmploymentDB()
+            insertCyaData(employmentUserData(isPrior = true, cyaModel("name", hmrc = true, Some(benefits(fullTravelOrEntertainmentModel)))), userRequest)
+            urlPost(url(taxYearEOY), follow = false, welsh = user.isWelsh,
+              headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)), body = Map("amount" -> "9999999999999999999999999999"))
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          "has an BAD_REQUEST status" in {
+            result.status shouldBe BAD_REQUEST
+          }
+
+          titleCheck(get.expectedErrorTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(user.commonExpectedResults.expectedCaption)
+          textOnPageCheck(user.commonExpectedResults.previousExpectedContent + " " + get.expectedContent, contentSelector)
+          textOnPageCheck(user.commonExpectedResults.amountHint, hintTextSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          inputFieldValueCheck("9999999999999999999999999999", inputSelector)
+
+          buttonCheck(user.commonExpectedResults.continue, continueButtonSelector)
+          formPostLinkCheck(continueLink, continueButtonFormSelector)
+          welshToggleCheck(user.isWelsh)
+
+          errorSummaryCheck(get.maxAmountErrorText, expectedErrorHref)
+          errorAboveElementCheck(get.maxAmountErrorText)
+        }
+      }
+    }
+
+    "redirect to check employments benefits page when a valid form is submitted and a prior submission" when {
```

Comment:
It might be worth updating the amount with a value thats not 100, because from looking in the fullTravelOrEntertainmentModel it was already 100. So youre essentially updating 100 to 100.

Reply:
Thank you, I initially had another starting value, but as part of a rebase I updated it again.